test/functional/provider/define_spec.lua

```diff
@@ -333,6 +333,39 @@ local function function_specs_for(fn, sync, first_arg_factory, init)
           runx(sync, handler, on_setup)
         end)
       end)
+
+      describe('with range', function()
```

Comment:
I agree we should move away from the odd pattern in this test. It's come up before, but we keep adding more tests using the existing pattern. @snoe If you wouldn't mind, it would be preferrable to collapse the cases where a single `it()` is contained in a `describe()`, to just an `it()`.